### PR TITLE
fix resources not getting deleted for unique ns cron

### PR DESCRIPTION
**Pull request by pradeepitm12**
Issue: add unique value for schedule as annotation to a namespace.
It creates a cronjob in that ns.
But when the job is getting actually scheduled it does not find
tekton-pipeline-controller sa. Which gives error

Fix: Instead of creating cronjob in other ns create it in the
installation target ns. There the sa is already available.

Signed-off-by: Pradeep Kumar <pradkuma@redhat.com>

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/operator/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

### CODE REVIEW

**Comment by pradeepitm12** [comment]
cc: @vdemeester

File: pkg/reconciler/common/prune_test.go [hunk 1]
```diff
@@ -122,6 +123,16 @@ func TestCompleteFlowPrune(t *testing.T) {
 	if len(cronjobs.Items) != 2 {
 		assert.Error(t, err, "number of cronjobs not correct")
 	}
+	client.CoreV1().Namespaces().Update(context.TODO(), &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "ns-two"}}, metav1.UpdateOptions{})
```
Comment (vdemeester) [inline pkg/reconciler/common/prune_test.go:126]: According to the build error, we need to check this error 😉

**Review by vdemeester — APPROVED** [review]

File: pkg/reconciler/common/prune.go [hunk 2]
```diff
@@ -249,7 +250,24 @@ func (pruner *Pruner) createJobContainers(nsConfig *pruneConfigPerNS, ns string)
...
+func (pruner *Pruner) checkAndCreate(ctx context.Context, cronName, targetNs, uniquePruneNs, schedule string, pruneContainers []corev1.Container, oR v1.OwnerReference, listOpt string) error {
+	cronList, err := pruner.listCronJobs(ctx, targetNs, listOpt)
+	if err != nil {
+		return err
+	}
+	if len(cronList.Items) == 1 {
+		if err := pruner.deleteCronJob(ctx, cronList.Items[0].Name, targetNs); err != nil {
+			return err
+		}
+		return createCronJob(ctx, pruner.kc, cronName, targetNs, uniquePruneNs, schedule, pruneContainers, oR)
```
Comment (nikhil-thomas) [inline pkg/reconciler/common/prune.go:262]:
```suggestion
```
this line could be safely removed, as the next line will handle the create action for both cases

File: pkg/reconciler/common/prune.go [hunk 3]
```diff
@@ -249,7 +250,24 @@ func (pruner *Pruner) createJobContainers(nsConfig *pruneConfigPerNS, ns string)
...
+	}
+	if err := createCronJob(ctx, pruner.kc, cronName, targetNs, uniquePruneNs, schedule, pruneContainers, oR); err != nil {
+		return err
+	}
+	return nil
```
Comment (nikhil-thomas) [inline pkg/reconciler/common/prune.go:267]:
```suggestion
	if len(cronList.Items) == 1 {
		if err := pruner.deleteCronJob(ctx, cronList.Items[0].Name, targetNs); err != nil {
			return err
		}
	}
	return createCronJob(ctx, pruner.kc, cronName, targetNs, uniquePruneNs, schedule, pruneContainers, oR)
```
as the eventual create actions is same from the if block and the normal flow, let us keep only one statement for create.

Comment (nikhil-thomas) [reply pkg/reconciler/common/prune.go:267]: not critical, :technologist: :+1:

**Pull request merged**